test/export/test_export.py

```diff
@@ -2446,6 +2490,36 @@ def g(x):
         ):
             constraints = [dynamic_dim(inp_for_g, 0)]
 
+    def test_mark_and_auto_dynamic(self):
+        # for this use case, mark_dynamic() and AUTO should have same effect.
+        # check that same symbol gets allocated to both dims without raising constraint violation.
+        from torch.export.dynamic_shapes import DIM
+
+        AUTO, STATIC = DIM.AUTO, DIM.STATIC
+
+        class Foo(torch.nn.Module):
+            def forward(self, x, y):
+                torch._check(x.shape[0] == y.shape[0])
+                torch._check(x.shape[0] <= 64)
+                return x + 2, y + 2
+
+        inputs = (torch.randn(4, 4), torch.randn(4, 4))
+        ep_auto = torch.export.export(
+            Foo(), inputs, dynamic_shapes={"x": (AUTO, None), "y": (AUTO, None)}
+        )
+        torch._dynamo.mark_dynamic(inputs[0], 0)
```

Comment:
Can you say whether `mark_dynamic` only works on dims declared `AUTO` or does it work for anything? Any assertions you want to make here?

Reply:
Ah it'll work on anything